test/cql-pytest/conftest.py

```diff
@@ -221,3 +221,18 @@ def temp_workdir():
     """ Creates a temporary work directory, for the scope of a single test. """
     with tempfile.TemporaryDirectory() as workdir:
         yield workdir
+
+@pytest.fixture(scope="session")
+def has_tablets(cql, scylla_only):
+    with new_test_keyspace(cql, " WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'replication_factor': 1}") as keyspace:
+        return keyspace_has_tablets(cql, keyspace)
+
+@pytest.fixture(scope="function")
+def xfail_tablets(request, has_tablets):
+    if has_tablets:
+        request.node.add_marker(pytest.mark.xfail(reason='Test expected to fail with tablets experimental feature on'))
+
+@pytest.fixture(scope="function")
+def skip_with_tablets(request, has_tablets):
+    if has_tablets:
+        pytest.skip("test may crash with tablets enabled")
```

Comment:
Nice catch. In my own tests I only used test/cql-pytest/run, not test.py, so I didn't use that horrible "on-internal-error-causes-a-crash" mode I didn't see the need for skip. But yes, in test.py it's unforunately needed (https://github.com/scylladb/scylladb/issues/13352).

Reply:
These are all in the select_from_mutation_fragments.py, we can remove this fixture altogether once https://github.com/scylladb/scylladb/pull/16802 gets in.